src/ios/CameraSessionManager.m

```diff
@@ -167,6 +176,82 @@ - (void) torchSwitch:(NSInteger)torchState{
     }
 }
 
+- (void)switchToUltraWideCamera:(NSString*)cameraMode completion:(void (^)(BOOL success))completion {
+    if (![self deviceHasUltraWideCamera]) {
+        if (completion) {
+            completion(NO);
+        }
+        return;
+    }
+
+    dispatch_async(self.sessionQueue, ^{
+        if (@available(iOS 13.0, *)) {
+            AVCaptureDevice *ultraWideCamera;
+            if([cameraMode isEqualToString:@"default"]) {
+                ultraWideCamera = [self cameraWithPosition:self.defaultCamera captureDeviceType:AVCaptureDeviceTypeBuiltInWideAngleCamera];
+            } else {
+                ultraWideCamera = [self cameraWithPosition:self.defaultCamera captureDeviceType:AVCaptureDeviceTypeBuiltInUltraWideCamera];
+            }
+            if (ultraWideCamera) {
+                // Remove the current input
+                [self.session removeInput:self.videoDeviceInput];
+                
+                // Create a new input with the ultra-wide camera
+                NSError *error = nil;
+                AVCaptureDeviceInput *ultraWideVideoDeviceInput = [AVCaptureDeviceInput deviceInputWithDevice:ultraWideCamera error:&error];
+                
+                if (!error) {
+                    // Add the new input to the session
+                    if ([self.session canAddInput:ultraWideVideoDeviceInput]) {
+                        [self.session addInput:ultraWideVideoDeviceInput];
+                        self.videoDeviceInput = ultraWideVideoDeviceInput;
+                        __block AVCaptureVideoOrientation orientation;
+                        dispatch_sync(dispatch_get_main_queue(), ^{
+                            orientation = [self getCurrentOrientation];
+                        });
+                        [self updateOrientation:orientation];
+                        if (completion) {
+                            completion(YES);
+                        }
+                    } else {
+                        NSLog(@"Failed to add ultra-wide input to session");
+                        if (completion) {
+                            completion(NO);
+                        }
+                    }
+                } else {
+                    NSLog(@"Error creating ultra-wide device input: %@", error.localizedDescription);
+                    if (completion) {
+                        completion(NO);
+                    }
+                }
+            } else {
+                NSLog(@"Ultra-wide camera not found");
+                if (completion) {
+                    completion(NO);
+                }
+            }
+        } else {
+            // Fallback on earlier versions
+            if (completion) {
+                completion(NO);
+            }
+        }
```

Comment:
can we rework this with return statement for truthy part of the if(s) and at the end only one completion(NO)